examples/graphs/hld_path_composite_yosupo.rs

```diff
@@ -13,12 +15,12 @@ fn main() {
     let a = (0..n)
         .map(|_| {
             input! {
-                c: usize,
-                d: usize
+                c: u64,
+                d: u64
```

Comment:
one thing I like about `const MOD: u64 = 998_244_353;` is naturally you'll use `u64` for all mod operations

well first, I think we should use `u64` instead of `usize` for mod operations, so that we don't overflow on judges with 32 bit machines

But another question is should we use `u64` or `u32` for all mod operations? And here, a benefit of `u32` is less space. But you need so many casts just to multiple, that I think we should just use `u64`

Reply:
yeah I would opt for u64, forward thinking most judges will all be 64 bit machines anyway optimistically

Reply:
I would also find it hard to believe it would save any reasonable amount of time on a 32 bit machine with the necessary casts